File: backend/src/modules/floors/floor.controller.js
const floorService = require('./floor.service');
const siteVersionService = require('../siteVersions/siteVersion.service');

async function listByBuilding(req, res) {
  const floors = await floorService.listFloorsByBuilding(
    req.params.buildingId
  );
  res.json(floors);
}

async function createForBuilding(req, res) {
  const floor = await floorService.createFloor(
    req.params.buildingId,
    req.body
  );
  const siteId = floor.building?.siteId;
  if (siteId) {
    await siteVersionService.syncWorkingPayloadFromDb(siteId);
  }
  res.status(201).json(floor);
}

async function updateById(req, res) {
  const floor = await floorService.updateFloor(req.params.floorId, req.body);
  const siteId = floor.building?.siteId;
  if (siteId) {
    await siteVersionService.syncWorkingPayloadFromDb(siteId);
  }
  res.json(floor);
}

async function deleteById(req, res) {
  const existing = await floorService.deleteFloor(req.params.floorId);
  const siteId = existing.building?.siteId;
  if (siteId) {
    await siteVersionService.syncWorkingPayloadFromDb(siteId);
  }
  res.status(204).send();
}

module.exports = {
  listByBuilding,
  createForBuilding,
  updateById,
  deleteById,
};

File: server/src/main/java/com/legioncontrols/server/sites/FloorController.java
package com.legioncontrols.server.sites;

import com.legioncontrols.server.releases.ReleaseService;
import java.util.List;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

/** Java equivalent of backend/src/modules/floors/floor.controller.js. */
@RestController
public class FloorController {

    private final FloorService floorService;
    private final ReleaseService releaseService;

    public FloorController(FloorService floorService, ReleaseService releaseService) {
        this.floorService = floorService;
        this.releaseService = releaseService;
    }

    @GetMapping("/api/buildings/{buildingId}/floors")
    public List<FloorDto> listByBuilding(@PathVariable String buildingId) {
        return floorService.listFloorsByBuilding(buildingId);
    }

    @PostMapping("/api/buildings/{buildingId}/floors")
    public ResponseEntity<FloorDto> createForBuilding(@PathVariable String buildingId, @RequestBody FloorService.CreateFloorRequest request) {
        var created = floorService.createFloor(buildingId, request);
        releaseService.syncWorkingPayloadFromDb(created.siteId());
        return ResponseEntity.status(HttpStatus.CREATED).body(FloorDto.from(created.floor()));
    }

    @PatchMapping("/api/floors/{floorId}")
    public FloorDto updateById(@PathVariable String floorId, @RequestBody FloorService.UpdateFloorRequest request) {
        var updated = floorService.updateFloor(floorId, request);
        releaseService.syncWorkingPayloadFromDb(updated.siteId());
        return FloorDto.from(updated.floor());
    }

    @DeleteMapping("/api/floors/{floorId}")
    public ResponseEntity<Void> deleteById(@PathVariable String floorId) {
        var deleted = floorService.deleteFloor(floorId);
        releaseService.syncWorkingPayloadFromDb(deleted.siteId());
        return ResponseEntity.noContent().build();
    }
}
